### PR TITLE
doc: output alphabetic Kconfig options index

### PR DESCRIPTION
**Pull request by dbkinder**
Instead of a configuration options index in "discovered" order
during the walk of the Kconfig files, create the index in alphabetic
order.

jira: ZEP-2310

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

### CODE REVIEW

File: doc/scripts/genrest/genrest.py [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Prints a tree of all items in the configuration
 # vim: tabstop=4 shiftwidth=4 expandtab
 
+# instead of writing the index out in the order discovered,
```
Comment (inakypg) [inline doc/scripts/genrest/genrest.py]: make this the doc for variable itemIndex

File: doc/scripts/genrest/genrest.py [hunk 2]
```diff
@@ -90,4 +99,13 @@ def print_items(items, outdir, indent):
 f.write(".. list-table:: Configuration Options\n\n")
 conf = kconfiglib.Config(sys.argv[1])
 print_items(conf.get_top_level_items(), sys.argv[2],  0)
+
+# print_items creates separate .rst files for each configuration option as
+# well as a dictionary index of all these options. Print out the accumulated
+# config symbols index in alphabetic order
+
+sortedItems = sorted(itemIndex.keys())
```
Comment (inakypg) [inline doc/scripts/genrest/genrest.py]: compact to 

`for item in sorted(item_index):`

item_index, when iterated, evaluates to the keys, by default, so no need to call keys()

**Review by inakypg — CHANGES_REQUESTED** [review]

**Comment by dbkinder** [comment]
recommended changes made.

File: doc/scripts/genrest/genrest.py [hunk 3]
```diff
@@ -31,12 +32,23 @@ def print_items(items, outdir, indent):
...
                 done.append(var)
-                f.write("   * - :option:`%s`\n" %var)
+
+                # Instead of writing the index document with symbols listed
+                # in the order discovered, save up the items in a dictionary
+                # (itemIndex) and then write out the index doc with options in
+                # alphabetic order, together with the "prompt" (short
+                # description) from the Kconfig file.
+
+                #f.write("   * - :option:`%s`\n" %var)
```
Comment (mnkp) [inline doc/scripts/genrest/genrest.py]: Shouldn't we remove unused code rather than comment it out?

File: doc/scripts/genrest/genrest.py [hunk 4]
```diff
@@ -31,12 +32,23 @@ def print_items(items, outdir, indent):
...
                 if len(item.get_prompts()) > 0:
                     p = item.get_prompts()[0]
                 else:
                     p = ""
-                f.write("     - %s\n" %p)
+                #f.write("     - %s\n" %p)
```
Comment (mnkp) [inline doc/scripts/genrest/genrest.py]: Shouldn't we remove unused code rather than comment it out?

Comment (dbkinder) [inline doc/scripts/genrest/genrest.py]: good point.  I've cleaned that up now, and also updated comment to not talk about how it used to be, and rather how it is now.

**Comment by dbkinder** [comment]
review comments incorporated (shippable isn't returning)

**Comment by dbkinder** [comment]
seeing if close/re-open will give shipable testing a kick

**Comment by dbkinder** [comment]
@inakypg OK now?

**Review by inakypg — APPROVED** [review]

**Pull request merged**